src/GameData.ts

```diff
@@ -0,0 +1,265 @@
+        return await (await fetch(perkDataUrl)).json() as PerkData[];
+    }
+
+    public async loadItemData(): Promise<ItemData[]> {
+        const returnData: ItemData[] = [];
+
+        const itemDataUrl = "https://raw.communitydragon.org/latest/plugins/rcp-be-lol-game-data/global/default/v1/items.json";
+        const items = await (await fetch(itemDataUrl)).json();
+        items.forEach((c: ItemData) => returnData.push({ ...c }));
+        returnData.forEach((c: ItemData) => { c.to = []; c.from = []; });
+
+        items.forEach((i: any) => {
+            const item = returnData.filter(r => r.id === +i.id)[0];
+
+            i.from.forEach((f: number) => {
+                const otherItem = returnData.filter(r => r.id === f)[0];
+                if (otherItem !== undefined) {
+                    item.from.push(otherItem.name);
+                }
+            });
+
+            i.to.forEach((f: number) => {
+                const otherItem = returnData.filter(r => r.id === f)[0];
+                if (otherItem !== undefined) {
+                    item.to.push(otherItem.name);
+                }
+            });
+        });
+
+        return returnData;
+    }
+
+    public onLookup(message: Discord.Message, isAdmin: boolean, command: string, args: string[]) {
+        if (message.cleanContent.length === 0) {
+            let response = `I have info on the following categories; \`item\`,\`perk\`,\`champion\``;
+            response += `type !${command} {search_type} {search_term} to use it`;
+            message.channel.send(response);
+            return;
+        }
+
+        if (!["item", "perk", "champion"].some(i => i === args[0])) {
+            message.channel.send(`I'm sorry. I'm unable to parse the category \`${args[0]}\` at this moment. If you want it added, contact a guru`);
+            return;
+        }
+
+        if (args[0] === "item") {
+            message.channel.send(this.findItem(args.slice(1).join(" ")));
+            return;
+        }
+
+        if (args[0] === "perk") {
+            message.channel.send(this.findPerk(args.slice(1).join(" ")));
+            return;
+        }
+
+        if (args[0] === "champion") {
+            message.channel.send(this.findChampion(args.slice(1).join(" ")));
+            return;
+        }
+    }
+
+    public findItem(search: string): string {
+        if (!search) {
+            return `There are currently ${this.itemData.length} items in my lookup data!`;
+        }
+
+        const result = this.itemData.filter(c => Math.min(
+            ...[
+                levenshteinDistance(search, c.id.toString()),
+                levenshteinDistance(search, c.name),
+                levenshteinDistance(search, c.categories.toString()),
+            ]) < 1);
+
+        if (result.length === 0) {
+            return "Unable to find any data for that search term.";
+        }
+
+        if (result.length > 3) {
```

Comment:
If there's 1-3 items, you only return the first one and don't show an error. Either this should be 1 or you should return the other matches too.

Reply:
I would agree, show up to 3 and any more you would omit with a message like `..x more results found.`

Reply:
Is this resolved @stelar7?